doc/ansible/dev/dependent_watches.md

```diff
@@ -0,0 +1,33 @@
+# Dependent Watches
+This document describes the `watchDependentResources` option in [`watches.yaml`](https://github.com/operator-framework/operator-sdk/blob/0136d0ec3f916364a4f005895ef12aa80ab8f2de/doc/ansible/dev/advanced_options.md#watches-file-options) file. It delves into what dependent resources are, why the option is required, how it is achieved and finally gives an example.
```

Comment:
NIT @alaypatel07 shouldn't this link be described differently than `watches.yaml`?  perhaps `the watches file` or something like that?  I was expecting an example `watches.yaml` file when I followed the link.

Reply:
I have given the example later in the document, wanted to point readers on watchDependentRerources in watches file documentation. Thoughts @shawn-hurley @jmrodri ?

Also, I think it would be better to use relative links in the markdown!